azkaban-common/src/main/java/azkaban/project/FlowTriggerDependency.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+import azkaban.utils.Props;
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Dependency is an immutable class which holds
+ * all the necessary properties of a dependency.
+ */
+public class FlowTriggerDependency {
+
+  private final Props props;
+  private final String name;
+  private final String type;
+
+  public FlowTriggerDependency(final String name, final String type, final Props depProps) {
+    Preconditions.checkArgument(StringUtils.isNotEmpty(name));
+    Preconditions.checkArgument(StringUtils.isNotEmpty(type));
+    Preconditions.checkArgument(depProps != null);
+    this.name = name;
+    this.type = type;
+    this.props = new Props(depProps.getParent(), depProps);
+    //todo chengren311: validate per dependencyType: daliviewdepenency needs extra special check:
```

Comment:
daliviewdepenency doesn't belong to the OSS version.

Reply:
updated